rasa_nlu/classifiers/intent_threshold.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import os
+import io
+import json
+from builtins import map
+from typing import Any
+from typing import Dict
+from typing import Text
+import re
+
+
+import rasa_nlu.converters as converters
+from rasa_nlu.training_data import TrainingData
+
+from rasa_nlu.components import Component
+
+
+class IntentThreshold(Component):
+
+    name = "intent_threshold"
+
+    provides = ["intent"]
+
+    requires = ["intent"]
+
+    def __init__(self, intent_threshold=None):
+        # 'threshold' should be an attribute
+        self.intent_threshold = intent_threshold
+
+    def train(self, training_data, config, **kwargs):
+        self.intent_threshold = config["intent_threshold"]
+        print("threshold: {}".format(self.intent_threshold))   
+
+    def process(self, message, **kwargs):
+        # type: (Text) -> Dict[Text, Any]
+        # Explicitly set the threshold
+        # self.intent_threshold = 0.55
+        #print("intent: {}".format(message.get("intent")))
+        #print("threshold: {}".format(self.intent_threshold))
+        if message.get("intent").get("confidence", 1.0) < self.intent_threshold:
+            message.set("intent", {
+                    "name": "out_of_scope",
+                    "confidence": 1.0,
+                }, add_to_output=True)
+
+    def persist(self, model_dir):
+        # type: (Text) -> Dict[Text, Any]
+        file_name = self.name+".json"
+        full_name = os.path.join(model_dir, file_name)
+        with io.open(full_name, 'w') as f:
+            f.write(unicode(json.dumps({"intent_threshold": self.intent_threshold})))
```

Comment:
`unicode` doesn't exist in python 3. have a look how this is done in other components (e.g. `duckling_extractor.py`)